### ISSUE TITLE
Wrong Content-Type in 4.1.0

### ISSUE THREAD

**Issue by duk3luk3** [wopian/kitsu#86]
I just upgraded from 3.1.4 to 4.1.0.

Here's how I initialize kitsu:

```js
<script>
const Kitsu = require('kitsu');

const api = new Kitsu({
    baseURL: '{{config['PREFERRED_URL_SCHEME']}}://{{config['SERVER_NAME']}}/api',
//    headers: {
//        Authorization: 'foobar'
//    }
});
api.headers['Authorization'] = 'foobar';
```

And here's how I make a request:

```js
        api.update('containers', {
            id: container_id,
            status: container_action
        }).then(function(res) {
            console.log(res);
        })
            .catch((reason) => console.log(reason));
```

My requests should be sent with `application/vnd.api+json` Content-Type, but they aren't, no matter what I do it is sent with `application/json;charset=utf-8` which happens to be the Axios default.

So I suspect that something broke between 3.x and 4.x that causes Axios to no longer set / see the correct Content-Type.

Here is the libs I am using:

```
├─ kitsu@4.1.0
│  ├─ axios@^0.17.0
│  ├─ babel-runtime@^6.26.0
│  └─ pluralize@^7.0.0
```

**Comment by wopian** [comment]
I can't seem to replicate this - request headers correctly contained `application/vnd.api+json` for everything I've tried. 

Only change is that axios is now just imported and used directly instead of being added to the class and used internally. This shouldn't have any effect on the headers though - the `Accept` and `Content-Type` headers are added over everything else.

---

Also you'll want to add `resourceCase: 'snake'` to the initialize options - the default changed from snake_case to kebab-case in `4.x`

**Comment by duk3luk3** [comment]
Hmm, I'll debug into this more once I am less tired.

Is there a way to get the non-minified versions of axios and kitsu browserified with all its dependencies?

**Comment by wopian** [comment]
Actually - the headers were changed from `content-type` to `Content-Type` in `4.x` and _is_ dropping the content type header, but not the accept header (I may have been looking at the responses' content type header last night 😆). May be a [bug in Axios](https://github.com/axios/axios/blob/6e605016f03c59b5e0c9d2855deb3c5e6ec5bbfc/lib/defaults.js#L50) though.

You can use [Browserify](http://browserify.org) to bundle the dependencies into a single file.

**Comment by duk3luk3** [comment]
> You can use Browserify to bundle the dependencies into a single file.

Yes, but how do I do that with the non-minified versions?

**Comment by wopian** [comment]
You'll need to target one of the `lib` files as Browserify doesn't like ES6 imports - run `yarn build` then `yarn run browserify lib/kitsu.js -o src_bundle.js`

**Comment by duk3luk3** [comment]
OK, yep, that works, I have no idea how I managed not to try that variant during my attempts at randomly guessing how to get it right...

So I think I've figured out the problem - could have figured that out just by staring harder at the code instead of spending hours making it debuggable, but it appears that the axios api actually changed:

https://github.com/axios/axios/blob/master/README.md#axiospatchurl-data-config

```
axios.patch(url[, data[, config]])
```

kitsu does this:

https://github.com/wopian/kitsu/blob/6b0ee2b91ad19f4c131506a182b48171d9ee8a3c/src/index.js#L180-L183

But should do this:

```js
      const { data } = await axios.patch(url,
        {data: (await serialise.apply(this, [ model, body, 'PATCH' ])).data},
        {headers: headers}
      )
```

In my browserified js, this is what the working code looks like:

```js
        var _ref2 = yield axios.patch(url,
            {data: (yield serialise.apply(_this2, [model, body, 'PATCH'])).data},
          {headers: headers}
        );

        const data = _ref2.data;
```

If that analysis makes sense to you, I'll be happy to make a PR.

**Comment by wopian** [comment]
Fixed in `4.2.0`
